pandas/tests/tools/test_to_datetime.py

```diff
@@ -2773,3 +2773,22 @@ def test_to_datetime_monotonic_increasing_index(cache):
     result = to_datetime(times.iloc[:, 0], cache=cache)
     expected = times.iloc[:, 0]
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "series_length",
+    [40, start_caching_at, (start_caching_at + 1), (start_caching_at + 5)],
+)
+def test_to_datetime_cache_coerce_50_lines(series_length):
```

Comment:
```suggestion
def test_to_datetime_cache_coerce_50_lines_outofbounds(series_length):
```

Also could you test the `ignore` and `raise` options as well?